Install app deps always trigger rebuild

<!-- Which version of electron-builder are you using? -->
* **Version**: 7.10.0

<!-- What target are you building for? -->
* **Target**: Mac and Linux

Hello, I'm using electron-builder with native module (https://github.com/lyssdod/node-libtorrent). The module itself uses `node-pre-gyp` and has precompiled binaries. It installs flawlessly if I'm not using `electron-builder` directly (running `npm install` from `app` directory works). But if I do, it fails with

```
Error output:
module.js:457
    throw err;
    ^

Error: Cannot find module 'adm-zip'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
```

The fun fact is, this module (`adm-zip`) is specified in `node-libtorrent-ng`s `devDependencies` and is handled by regular `npm install` just fine. 

So my questions are:
1. Why `electron-builder` ignores default action of `node-gyp-rebuild` despite binaries are available?
2. Is there a way to debug 'install-app-deps' step?

Thanks!

--- a/docs/Options.md
+++ b/docs/Options.md
@@ -80,6 +80,7 @@ Don't customize paths to background and icon, — just follow conventions.
 | compression | <a name="BuildMetadata-compression"></a>The compression level, one of `store`, `normal`, `maximum` (default: `normal`). If you want to rapidly test build, `store` can reduce build time significantly.
 | afterPack | <a name="BuildMetadata-afterPack"></a>*programmatic API only* The function to be run after pack (but before pack into distributable format and sign). Promise must be returned.
 | npmRebuild | <a name="BuildMetadata-npmRebuild"></a>*two package.json structure only* Whether to [rebuild](https://docs.npmjs.com/cli/rebuild) native dependencies (`npm rebuild`) before starting to package the app. Defaults to `true`.
+| npmSkipBuildFromSource | <a name="BuildMetadata-npmSkipBuildFromSource"></a>*two package.json structure only* Whether to omit using [--build-from-source](https://github.com/mapbox/node-pre-gyp#options) flag when installing app native deps. Defaults to `false`.
 | nodeGypRebuild | <a name="BuildMetadata-nodeGypRebuild"></a>Whether to execute `node-gyp rebuild` before starting to package the app. Defaults to `false`.
 | electronDist | <a name="BuildMetadata-electronDist"></a>The path to custom Electron build (e.g. `~/electron/out/R`). Only macOS supported, file issue if need for Linux or Windows.
 

--- a/src/metadata.ts
+++ b/src/metadata.ts
@@ -202,6 +202,11 @@ export interface BuildMetadata {
   readonly npmRebuild?: boolean
 
   /*
+   *two package.json structure only* Whether to omit using [--build-from-source](https://github.com/mapbox/node-pre-gyp#options) flag when installing app native deps. Defaults to `false`.
+   */
+  readonly npmSkipBuildFromSource?: boolean
+
+  /*
    Whether to execute `node-gyp rebuild` before starting to package the app. Defaults to `false`.
    */
   readonly nodeGypRebuild?: boolean

--- a/src/packager.ts
+++ b/src/packager.ts
@@ -235,7 +235,8 @@ export class Packager implements BuildInfo {
         log("Skip app dependencies rebuild because npmRebuild is set to false")
       }
       else if (platform.nodeName === process.platform) {
-        await installDependencies(this.appDir, this.electronVersion, Arch[arch], (await statOrNull(path.join(this.appDir, "node_modules"))) == null ? "install" : "rebuild")
+        let doSourceBuild = !(this.devMetadata.build.npmSkipBuildFromSource === true)
+        await installDependencies(this.appDir, this.electronVersion, Arch[arch], (await statOrNull(path.join(this.appDir, "node_modules"))) == null ? "install" : "rebuild", doSourceBuild)
       }
       else {
         log("Skip app dependencies rebuild because platform is different")

--- a/src/util/util.ts
+++ b/src/util/util.ts
@@ -16,8 +16,8 @@ export const debug7z = debugFactory("electron-builder:7z")
 
 const DEFAULT_APP_DIR_NAMES = ["app", "www"]
 
-export function installDependencies(appDir: string, electronVersion: string, arch: string = process.arch, command: string = "install"): BluebirdPromise<any> {
-  return task(`${(command === "install" ? "Installing" : "Rebuilding")} app dependencies for arch ${arch} to ${appDir}`, spawnNpmProduction(command, appDir, getGypEnv(electronVersion, arch)))
+export function installDependencies(appDir: string, electronVersion: string, arch: string = process.arch, command: string = "install", fromSource: boolean = true): BluebirdPromise<any> {
+  return task(`${(command === "install" ? "Installing" : "Rebuilding")} app dependencies for arch ${arch} to ${appDir}`, spawnNpmProduction(command, appDir, getGypEnv(electronVersion, arch), fromSource))
 }
 
 export function getGypEnv(electronVersion: string, arch: string): any {

--- a/src/util/util.ts
+++ b/src/util/util.ts
@@ -32,15 +32,18 @@ export function getGypEnv(electronVersion: string, arch: string): any {
   })
 }
 
-export function spawnNpmProduction(command: string, appDir: string, env?: any): BluebirdPromise<any> {
+export function spawnNpmProduction(command: string, appDir: string, env?: any, forceBuild: boolean = true): BluebirdPromise<any> {
   let npmExecPath = process.env.npm_execpath || process.env.NPM_CLI_JS
-  const npmExecArgs = [command, "--production", "--build-from-source", "--cache-min", "999999999"]
+  let npmExecArgs = [command, "--production", "--cache-min", "999999999"]
   if (npmExecPath == null) {
     npmExecPath = process.platform === "win32" ? "npm.cmd" : "npm"
   }
   else {
     npmExecArgs.unshift(npmExecPath)
     npmExecPath = process.env.npm_node_execpath || process.env.NODE_EXE || "node"
+  }
+  if (forceBuild) {
+    npmExecArgs.push("--build-from-source")
   }
 
   return spawn(npmExecPath, npmExecArgs, {

